test/elixir_analyzer/exercise_test/common_checks/function_capture_test.exs

```diff
@@ -0,0 +1,379 @@
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.FunctionCaptureTest do
+  use ExUnit.Case
+
+  alias ElixirAnalyzer.ExerciseTest.CommonChecks.FunctionCapture
+  alias ElixirAnalyzer.Comment
+  alias ElixirAnalyzer.Constants
+
+  @comment %Comment{
+    type: :informative,
+    comment: Constants.solution_use_function_capture(),
+    name: Constants.solution_use_function_capture()
+  }
+
+  test "using valid notation for several cases" do
+    code =
+      quote do
+        defmodule Capture do
+          import Float, only: [ratio: 1, pow: 2]
+
+          def captured(input) do
+            input
+            |> Enum.map(&ratio/1)
+            |> Enum.map(&Float.round/1)
+            |> Enum.map(&:math.sin/1)
+            |> Enum.zip_with(input, &pow/2)
+            |> Enum.zip_with(input, &Float.ceil/2)
+            |> Enum.zip_with(input, &:math.atan2/2)
+          end
+
+          def legit(input) do
+            input
+            |> Enum.reduce([], fn n, acc -> div(acc, n) end)
+            |> Enum.reduce([], &div(&2, &1))
+            |> Enum.reduce([], fn n, acc -> String.pad_leading(n, acc, "42") end)
+            |> Enum.reduce([], &String.pad_leading(&1, &2, "42"))
+            |> Enum.map(fn x -> String.split(x, ";") end)
+            |> Enum.map(&String.split(&1, ";"))
```

Comment:
I thought of two more corner cases that already work as expected, but it might be nice to add them to the tests:
```suggestion
            |> Enum.map(&String.split(&1, ";"))
            # behavior modified by using guards
            |> Enum.map(fn x when is_binary(x) -> String.downcase(x) end)
            # behavior modified by using pattern-matching
            |> Enum.map(fn {a, b} = x -> inspect(x) end)
            |> Enum.map(fn x = {a, b} -> inspect(x) end)
```

Reply:
Good idea, they were in my head but not on the page.